src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -604,17 +610,18 @@ class AnonymousFunctions
                 Dim descriptorsMap = diagnosticService.GetDiagnosticDescriptors(project)
                 Assert.Equal(1, descriptorsMap.Count)
 
-                ' Ask for document diagnostics multiple times, and verify compilation diagnostics are not reported.
+                ' Ask for document diagnostics multiple times, and verify compilation diagnostics are reported.
                 Dim document = project.Documents.Single()
+
                 Dim fullSpan = document.GetSyntaxRootAsync().WaitAndGetResult(CancellationToken.None).FullSpan
                 Dim diagnostics = diagnosticService.GetDiagnosticsForSpanAsync(document, fullSpan, CancellationToken.None).WaitAndGetResult(CancellationToken.None)
-                Assert.Equal(0, diagnostics.Count())
+                Assert.Equal(1, diagnostics.Count())
```

Comment:
Basically this analyzer reports 2 diagnostics, one with location and one without location (see projectDiagnostics below). We should verify that the single diagnostic returned by GetDiagnosticsForSpanAsync is same as withDocumentLocationDiagnostic.